jupyter/README.md

```diff
@@ -0,0 +1,138 @@
+
+      ```
+      $ ipython
+      In [1]: from notebook.auth import passwd
+      In [2]: passwd()
+      Enter password: ****
+      Verify password: ****
+      Out[2]: 'your_hashed_password'
+      ```
+
+      Add hashed password to your notebook configuration file.
+
+      Find the `c.NotebookApp.password` field in `.jupyter/jupyter_notebook_config.py`, which might be commented out. Modify the line as follows -
+      ```
+      c.NotebookApp.password = u'your_hashed_password'
+      ```
+
+    Automatic password setup will store the hash in `jupyter_notebook_config.json` while this method stores the hash in `jupyter_notebook_config.py`. The `.json` configuration options take precedence over the `.py` one, thus the manual password may not take effect if the `.json` file has a password set.
+
+  - Use SSL for encrypted communication
+
+    Generate a self signed certificate and key.
+
+    ```
+    openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout mykey.key -out mycert.pem
+    ```
+
+    You can start the notebook to communicate via a secure protocol mode by setting the certfile option to your self-signed certificate, i.e. mycert.pem, with the command:
+    ```
+    jupyter notebook --certfile=mycert.pem --keyfile mykey.key
+    ```
+
+  - Running a public notebook server
+
+    In the `~/.jupyter` directory, edit the notebook config file, `jupyter_notebook_config.py`. By default, the notebook config file has all fields commented out. The minimum set of configuration options that you should uncomment and edit in `jupyter_notebook_config.py` is the following:
+
+    ```
+    # Set options for certfile, ip, password, and toggle off
+    # browser auto-opening
+    c.NotebookApp.certfile = u'/absolute/path/to/your/certificate/mycert.pem'
+    c.NotebookApp.keyfile = u'/absolute/path/to/your/certificate/mykey.key'
+    # Set ip to '*' to bind on all interfaces (ips) for the public server
+    c.NotebookApp.ip = '*'
+    c.NotebookApp.password = u'sha1:bcd259ccf...<your hashed password here>'
+    c.NotebookApp.open_browser = False
+
+    # It is a good idea to set a known, fixed port for server access
+    c.NotebookApp.port = 9999
+    ```
+
+    You can then start the notebook as follows  -
+
+    1. Connect to sverdrup and open up interactive session on compute node. A typical command to do that is:
+    ```
+    srun -t 1440 -N 1 --ntasks 4 --cpus-per-task=7 --pty bash -i
+    ```
+
+    2. Unset this variable to avoid some permissions issues. On a terminal in the compute node type:
+    ```
+    unset XDG_RUNTIME_DIR
+    ```
+
+    3. Activate your conda environment on the compute node. Use this command to activate the environment `py38` (see Prerequisites) and add it to your jupyter kernel.
+    ```
+    conda activate py38
+    python -m ipykernel install --user --name=py38
+    ```
+
+    4. Run `open_notebook_on_sverdrup.sh`
```

Comment:
Super nitpicky ... would you be able to swap the `raw text` for hyperlinks for files that exist? This is a move I've only started doing in my readme's recently ... this can be done with:
```
[open_notebook_on_sverdrup.sh](open_notebook_on_sverdrup.sh)
```

Reply:
Yep, I have already done this while referring to py38.yaml. I will do this too!